File: Labtwo/C#/pr1.cs
using System;
using System.Text.RegularExpressions;
using System.Collections.Generic;

class Program
{
    // Функция поиска совпадений по регулярному выражению
    static List<string> FindPattern(string input, Regex pattern)
    {
        List<string> results = new List<string>();
        // Ищем все совпадения
        foreach (Match match in pattern.Matches(input))
        {
            results.Add(match.Value);
        }
        return results;
    }

    static void Main()
    {
        // Компилируем регулярное выражение
        Regex pattern = new Regex(@"1?10+1");
        
        Console.WriteLine("Write your string->");
        string input = Console.ReadLine();
        
        // Получаем и выводим результаты
        List<string> results = FindPattern(input, pattern);
        foreach (string res in results)
        {
            Console.WriteLine(res);
        }
    }
}

File: Labtwo/JS/pr1.js
// Функция поиска совпадений по регулярному выражению
function findPattern(str, pattern) {
    let matches = [];
    // Используем метод matchAll для поиска всех совпадений
    let match;
    while ((match = pattern.exec(str)) !== null) {
        matches.push(match[0]);
    }
    return matches;
}

console.log("Write your string->");
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('', input => {
    // Регулярное выражение для поиска
    const pattern = /1?10+1/g;
    
    // Получаем и выводим результаты
    const results = findPattern(input, pattern);
    results.forEach(res => console.log(res));
    
    readline.close();
});